rflx/model.py

```diff
@@ -1033,7 +1050,9 @@ def check_message_references(name: str, messages: Mapping[str, AbstractMessage])
 
 def qualified_literals(types: Mapping[Field, Type], package: str) -> Set[str]:
     return {
-        l if t.package == package or t.package == BUILTINS_PACKAGE else f"{t.package}.{l}"
+        l
+        if t.package == package or t.package == BUILTINS_PACKAGE or t.is_embedded
```

Comment:
In general I have a bad feeling about this `is_embedded` flag. I also don't think checking for `is_embedded` is the correct solution here. The actual problem is that when a message is embedded from a different package B which contains a condition with a literal of package B, then the merged message will still contain a condition with a literal of package B that is not prefixed. To fix this issue all literals in all conditions need to be prefixed with B.